File: frontend/src/types/Category.ts
import type { Transaction } from "./Transaction";

export const CategoryType = {
  Expense: "Expense",
  Income: "Income",
  Both: "Both"
} as const

export type CategoryType = typeof CategoryType[keyof typeof CategoryType]

export interface Category {
  id: number;
  description: string;
  purpose: CategoryType;
  transactions: Transaction[];
}

export const categoryLabels: Record<CategoryType, string> = {
  Expense: "Despesa",
  Income: "Receita",
  Both: "Ambas"
};

File: backend-web-api/Models/Category.cs
using backend_web_api.Enums;

namespace backend_web_api.Models
{
    public class Category
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public CategoryPurpose Purpose { get; set; }
        public List<Transaction> Transactions { get; set; } = new();
    }
}